qiskit/transpiler/passes/routing/sabre_swap.py

```diff
@@ -341,11 +342,19 @@ def empty_dag(node, block):
         return out
 
     def apply_inner(out_dag, current_layout, qubit_indices_inner, result, id_to_node):
-        for node_id in result.node_order:
+        if isinstance(result, SabreResult):
```

Comment:
You should be able to skip this check if you turn `block_result.result` into a tuple before this call.

https://github.com/Qiskit/qiskit-terra/pull/10652/files#diff-7d05140621ea52112a754c580b8d0056ed6cd8dcf44c7970ced6e2566959cc56R368-R374

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/10652/commits/bd734e216b9d23f7cb42bf50fc6672f0b70137a3